internal/ordered/map.go

```diff
@@ -301,8 +310,10 @@ func (m *Map[K, V]) UnmarshalJSON(b []byte) error {
 // UnmarshalYAML unmarshals a YAML mapping node into this map. It only supports
 // K = string. Where yaml.v3 typically infers map[string]any for unmarshaling
 // mappings into any, this method chooses *Map[string, any] instead.
+// If V = *yaml.Node, then the value nodes are not decoded. This is useful for
+// a shallow unmarshaling step.
 func (m *Map[K, V]) UnmarshalYAML(n *yaml.Node) error {
-	om, ok := (any)(m).(*Map[string, V])
```

Comment:
i'm commenting on the old diff here, but is `(any)(m)` the same as `any(m)`? it's just a regular type conversion?

Reply:
Yeah, I can't remember exactly but I probably had `()` in there while I was putting an older revision together.

Reply:
TIL that's even allowed!